lib/_http_client.js

```diff
@@ -22,6 +22,9 @@ function ClientRequest(options, cb) {
 
   if (typeof options === 'string') {
     options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
   } else {
     options = util._extend({}, options);
```

Comment:
Won't that default to localhost again? Isn't that what this is trying to prevent? I don't understand.